harvest/api/kraken.py

```diff
@@ -0,0 +1,292 @@
+# Builtins
+import yaml
+import datetime as dt
+from logging import critical, error, info, warning, debug
+from typing import Any, Dict, List, Tuple
+
+# External libraries
+import krakenex
+import pandas as pd
+
+# Submodule imports
+from harvest.api._base import API
+from harvest.utils import *
+
+class Kraken(API):
+
+    interval_list = ['1MIN', '5MIN', '15MIN', '30MIN', '1HR', '4HR', '1DAY', '7DAY', '15DAY']
+    crypto_ticker_to_kraken_names = {
+        'BTC': 'XXBT',     'ETH': 'XETH',      'ADA': 'ADA',
+        'USDT': 'USDT',    'XRP': 'XXRP',      'SOL': 'SOL',
+        'DOGE': 'XDG',    'DOT': 'DOT',       'USDC': 'USDC',
+        'UNI': 'UNI',      'LTC': 'XLTC',      'LINK': 'LINK',
+        'BCH': 'BCH',      'FIL': 'FIL',       'MATIC': 'MATIC',
+        'WBTC': 'WBTC',    'ETC': 'XETC',      'XLM': 'XXLM',
+        'TRX': 'TRX',      'DAI': 'DAI',       'EOS': 'EOS',
+        'ATOM': 'ATOM',    'AAVE': 'AAVE',     'XMR': 'XXMR',
+        'AXS': 'AXS',      'GRT': 'GRT',       'XTZ': 'XXTZ',
+        'ALGO': 'ALGO',    'MKR': 'MKR',       'KSM': 'KSM',
+        'WAVE': 'WAVE',    'COMP': 'COMP',     'DASH': 'DASH',
+        'CHZ': 'CHZ',      'ZEC': 'XZEC',      'MANA': 'MANA',
+        'ENJ': 'ENJ',      'SUSHI': 'SUSHI',   'YFI': 'YFI',
+        'QTUM': 'QTUM',    'FLOW': 'FLOW',     'SNX': 'SNX',
+        'BAT': 'BAT',      'SC': 'SC',         'ICX': 'ICX',
+        'PERP': 'PERP',    'BNT': 'BNT',       'OMG': 'OMG',
+        'CRV': 'CRV',      'ZRX': 'ZRX',       'NANO': 'NANO',
+        'ANKR': 'ANKR',    'SAND': 'SAND',     'REN': 'REN',
+        'KAVA': 'KAVA',    'MINA': 'MINA',     '1INCH': '1INCH',
+        'GHST': 'GHST',    'ANT': 'ANT',       'REP': 'XREP',
+        'REPV2': 'XREPV2', 'BADGER': 'BADGER', 'BAL': 'BAL',
+        'BAND': 'BAND',    'CTSI': 'CTSI',     'CQT': 'CQT',
+        'EWT': 'EWT',      'MLN': 'XMLN',      'ETH2': 'ETH2',
+        'GNO': 'GNO',      'INJ': 'INJ',       'KAR': 'KAR',
+        'KEEP': 'KEEP',    'KNC': 'KNC',       'LSK': 'LSK',
+        'LTP': 'LTP',      'LRC': 'LRC',       'MIR': 'MIR',
+        'OCEAN': 'OCEAN',  'PAXG': 'PAXG',     'RARI': 'RARI',
+        'REN': 'REN',      'XRP': 'XXRP',      'SRM': 'SRM',
+        'STORJ': 'STORJ',  'TBTC': 'TBTC',     'OGN': 'OGN',
+        'OXT': 'OXT'
+    }
```

Comment:
These are all the cryptos that kraken supports, if any new ones are added then this dictionary needs to be updated. This is to convert from the normal ticker name to the ticker that kraken uses in its api.